synapseclient/models/wiki.py

```diff
@@ -1 +1,1278 @@
+        if not self.id:
+            raise ValueError("Must provide id to get attachment handles.")
+
+        return await get_attachment_handles(
+            owner_id=self.owner_id,
+            wiki_id=self.id,
+            wiki_version=self.wiki_version,
+            synapse_client=synapse_client,
+        )
+
+    @otel_trace_method(
+        method_to_trace_name=lambda self, **kwargs: f"Get_Attachment_URL: Owner ID {self.owner_id}, Wiki ID {self.id}, File Name {kwargs['file_name']}"
+    )
+    async def get_attachment_async(
+        self,
+        *,
+        file_name: str,
+        download_file: bool = True,
+        download_location: Optional[str] = None,
+        synapse_client: Optional["Synapse"] = None,
+    ) -> Union[str, None]:
+        """
+        Download the wiki page attachment to a local file or return the URL.
+
+        Arguments:
+            file_name: The name of the file to get. The file name can be in either non-gzipped or gzipped format.
+            download_file: Whether associated files should be downloaded. Default is True.
+            download_location: The directory to download the file to. Required if download_file is True.
+            synapse_client: Optionally provide a Synapse client.
+        Returns:
+            If download_file is True, the attachment file will be downloaded to the download_location. Otherwise, the URL will be returned.
+        Raises:
+            ValueError: If owner_id or id is not provided.
+
+        Example: Get the attachment URL for a wiki page
+            This example shows how to get the attachment file or URL for a wiki page.
+            attachment_file_or_url = await WikiPage(owner_id=project.id, id=wiki_page.id).get_attachment_async(file_name="attachment.txt", download_file=False)
+            print(f"Attachment URL: {attachment_file_or_url}")
+
+        Example: Download the attachment file for a wiki page
+            This example shows how to download the attachment file for a wiki page.
+            attachment_file_path = await WikiPage(owner_id=project.id, id=wiki_page.id).get_attachment_async(file_name="attachment.txt", download_file=True, download_location="~/temp")
+            print(f"Attachment file path: {attachment_file_path}")
+        """
+        if not self.owner_id:
+            raise ValueError("Must provide owner_id to get attachment URL.")
+        if not self.id:
+            raise ValueError("Must provide id to get attachment URL.")
+        if not file_name:
+            raise ValueError("Must provide file_name to get attachment URL.")
+
+        client = Synapse.get_client(synapse_client=synapse_client)
+        # check if the file_name is in gzip format or image format
+        if (
+            not file_name.endswith(".gz")
+            and not file_name.endswith(".png")
+            and not file_name.endswith(".jpg")
+            and not file_name.endswith(".jpeg")
```

Comment:
nit: This condition is re-used in more than 1 location. It should be extracted to a single function that is re-used in both places to prevent a possibility that they get out of sync with one another.